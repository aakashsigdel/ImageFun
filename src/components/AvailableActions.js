import React from 'react';

import './Actions.scss';

const AvailableActions = props =>
  <div className="available-actions-container">
    {props.availableActions.map(item =>
      <div key={item.id}
        className="action"
        onClick={() => {props.pushAction(item)}}
      >
        {item.name}
      </div>
    )}
  </div>;

export default AvailableActions;
