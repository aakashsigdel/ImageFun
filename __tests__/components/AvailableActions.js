import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';

import AvailableActions from '../../src/components/AvailableActions';

import renderer from 'react-test-renderer';

describe('<AvailableActions /> component', () => {
  it('should render <AvailableActions /> correctly', () => {
    const tree = renderer.create(<AvailableActions availableActions={[]} />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call pushActions onClick', () => {
    const pushAction = sinon.spy();
    const wrapper = shallow(
      <AvailableActions
        availableActions={[{id: 1, action: 'roatate', name: 'Rotate'}]}
        pushAction={pushAction}
      />);
    wrapper.find('.action').simulate('click');

    expect(pushAction.calledOnce).toEqual(true);
  })
});



