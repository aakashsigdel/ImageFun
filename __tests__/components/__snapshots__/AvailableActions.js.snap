exports[`<AvailableActions /> component should render <AvailableActions /> correctly 1`] = `
<div
  className="available-actions-container" />
`;
